### PR TITLE
[TextFields] New controller api for leading and trailing colors.

### LINKED ISSUE #1782: [TextField] How to change color in leadingUnderlineLabel.textColor of TextField?
I do it like:
`self.myTextField.leadingUnderlineLabel.TextColor = [UIColor greenColor]; `
But nothing changes, the inscription remains black. How to change its color?

### PR DESCRIPTION
**Pull request by willlarche**
Includes tests and examples.

Closes #1782

### CODE REVIEW

File: components/TextFields/src/MDCTextInputControllerDefault.m [hunk 1]
```diff
@@ -865,6 +883,32 @@ - (BOOL)isDisplayingErrorText {
...
 
+- (UIColor *)leadingUnderlineLabelTextColor {
+  return _leadingUnderlineLabelTextColor ? _leadingUnderlineLabelTextColor :
+      [[self class] leadingUnderlineLabelTextColorDefault];
+}
+
+- (void)setLeadingUnderlineLabelTextColor:(UIColor *)leadingUnderlineLabelTextColor {
+  if (_leadingUnderlineLabelTextColor != leadingUnderlineLabelTextColor) {
+    _leadingUnderlineLabelTextColor = leadingUnderlineLabelTextColor ? leadingUnderlineLabelTextColor :
+        [[self class] leadingUnderlineLabelTextColorDefault];
```
Comment (romoore) [inline components/TextFields/src/MDCTextInputControllerDefault.m]: Do we want to use the "class property" dot syntax here?

File: components/TextFields/src/MDCTextInputControllerFullWidth.m [hunk 2]
```diff
@@ -167,8 +179,12 @@ - (void)encodeWithCoder:(NSCoder *)aCoder {
...
                 forKey:MDCTextInputControllerFullWidthInlinePlaceholderColorKey];
+  [aCoder encodeObject:self.leadingUnderlineLabelTextColor
+                forKey:MDCTextInputControllerFullWidthLeadingUnderlineLabelTextColor];
   [aCoder encodeConditionalObject:self.textInput
                            forKey:MDCTextInputControllerFullWidthTextInputKey];
+  [aCoder encodeObject:self.trailingUnderlineLabelTextColor
+                forKey:MDCTextInputControllerFullWidthTrailingUnderlineLabelTextColor];
 }
 
 - (instancetype)copyWithZone:(NSZone *)zone {
```
Comment (romoore) [inline components/TextFields/src/MDCTextInputControllerFullWidth.m:190]: Should this call [super copyWithZone:zone] to let NSObject do something?

File: components/TextFields/src/MDCTextInputControllerFullWidth.m [hunk 3]
```diff
@@ -529,6 +548,22 @@ - (BOOL)isDisplayingErrorText {
   return self.errorText != nil;
 }
 
+- (UIColor *)leadingUnderlineLabelTextColor {
+  return [UIColor clearColor];
```
Comment (romoore) [inline components/TextFields/src/MDCTextInputControllerFullWidth.m]: document why this is clear?

File: components/TextFields/tests/unit/TextFieldControllerClassPropertiesTests.swift [hunk 4]
```diff
@@ -56,6 +61,15 @@ class TextFieldControllerClassPropertiesTests: XCTestCase {
                    MDCPalette.blue.accent700)
     XCTAssertEqual(MDCTextInputControllerDefault.normalColorDefault, .lightGray)
     XCTAssertEqual(MDCTextInputControllerDefault.underlineViewModeDefault, .whileEditing)
+    XCTAssertEqual(MDCTextInputControllerDefault.leadingUnderlineLabelTextColorDefault,
+                   UIColor(white: 0, alpha: CGFloat(Float(0.54))))
```
Comment (romoore) [inline components/TextFields/tests/unit/TextFieldControllerClassPropertiesTests.swift:65]: `(CGFloat)0.54` is possible here

**Review by romoore — APPROVED** [review]

File: components/TextFields/src/MDCTextInputControllerFullWidth.m (hunk 2 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerFullWidth.m:190]: Turns out that's not a thing. NSObject doesn't implement NSCopying.

Comment (romoore) [reply components/TextFields/src/MDCTextInputControllerFullWidth.m:190]: Makes sense. Thanks for checking. 👍

File: components/TextFields/tests/unit/TextFieldControllerClassPropertiesTests.swift (hunk 4 above)
Comment (willlarche) [reply components/TextFields/tests/unit/TextFieldControllerClassPropertiesTests.swift:65]: No ;( Swift not having it.

File: components/TextFields/src/MDCTextInputControllerFullWidth.m (hunk 3 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerFullWidth.m]: Done!

**Pull request merged**